code/game/machinery/computer/atmos_control.dm

```diff
@@ -44,14 +44,13 @@
 	frequency = new_frequency
 	radio_connection = SSradio.add_object(src, frequency, RADIO_ATMOSIA)
 
-/obj/machinery/air_sensor/initialize()
+/obj/machinery/air_sensor/Initialize()
+	..()
 	set_frequency(frequency)
 
 /obj/machinery/air_sensor/New()
-	..()
 	SSair.atmos_machinery += src
-	if(SSradio)
-		set_frequency(frequency)
+	..()

```

Comment:
why did you move this to the bottom? It's good practice in new to always have the parent call first, because that ensures you have a loc and other vars before your own code is running